src/object/cli_obj.c

```diff
@@ -4332,11 +4538,17 @@ obj_reasb_io_fini(struct obj_auxi_args *obj_auxi, bool retry)
 	obj_update_sgls_free(obj_auxi);
 	obj_reasb_req_fini(&obj_auxi->reasb_req, obj_auxi->iod_nr);
 
-	/* zero it as user might reuse/resched the task, for
-	 * example the usage in dac_array_set_size().
-	 */
-	if (!retry)
+	if (!retry) {
+		if (obj_auxi->tx_local) {
```

Comment:
hmm, probably it is better to discuss if we really need this tx_local, really concern this kind of extra case.

Reply:
sure, we can discuss together with @Nasf-Fan next time. It is easy to remove just a few LOC.